graphene_django/views.py

```diff
@@ -286,64 +286,80 @@ def parse_body(self, request):
 
         return {}
 
-    def execute_graphql_request(
-        self, request, data, query, variables, operation_name, show_graphiql=False
-    ):
+    def execute_graphql_request(self, request, data, query, variables, operation_name):
         if not query:
-            if show_graphiql:
-                return None
             raise HttpError(HttpResponseBadRequest("Must provide query string."))
 
         try:
             document = parse(query)
         except Exception as e:
             return ExecutionResult(errors=[e])
 
-        if request.method.lower() == "get":
-            operation_ast = get_operation_ast(document, operation_name)
-            if operation_ast and operation_ast.operation != OperationType.QUERY:
-                if show_graphiql:
-                    return None
+        operation_ast = get_operation_ast(document, operation_name)
 
-                raise HttpError(
-                    HttpResponseNotAllowed(
-                        ["POST"],
-                        "Can only perform a {} operation from a POST request.".format(
-                            operation_ast.operation.value
-                        ),
-                    )
+        if not operation_ast:
+            ops_count = len(
+                [
+                    x
+                    for x in document.definitions
+                    if isinstance(x, OperationDefinitionNode)
+                ]
+            )
+            if ops_count > 1:
+                op_error = (
+                    "Must provide operation name if query contains multiple operations."
                 )
-        try:
-            extra_options = {}
-            if self.execution_context_class:
-                extra_options["execution_context_class"] = self.execution_context_class
+            elif operation_name:
+                op_error = f"Unknown operation named '{operation_name}'."
+            else:
+                op_error = "Must provide a valid operation."
+
+            return ExecutionResult(errors=[GraphQLError(op_error)])
```

Comment:
If `operation_ast` is not found it will just fail eventually anyway. Throw it here to simplify other validation checks like for the `GET` request and for the transaction checks for mutation ops.